### PR TITLE
Enable pylint and other linting rules in ruff

### PR DESCRIPTION
**Pull request by markbandstra**
This PR enables the following [rulesets](https://docs.astral.sh/ruff/rules/) in [`ruff`](https://docs.astral.sh/ruff/):

  - `SIM`: [flake8-simplify](https://pypi.org/project/flake8_simplify/) rules to simplify code
  - `PL`: [pylint](https://pypi.org/project/pylint/) rules to "check[] for errors, enforce[] a coding standard, look[] for [code smells](https://martinfowler.com/bliki/CodeSmell.html), and [] make suggestions about how the code could be refactored"
  - `FURB`: [refurb](https://pypi.org/project/refurb/), "for refurbishing and modernizing Python codebases"

I thought these made useful improvements and would be helpful for maintaining our code going forward.

### CODE REVIEW

File: becquerel/core/fitting.py [hunk 1]
```diff
@@ -1440,8 +1435,8 @@ def custom_plot(
             # Add info about the ROI and units
             if self.roi:
                 s += "ROI: [{:.3f}, {:.3f}]\n".format(*self.roi)
-            s += "X units: {:s}\n".format(self.xmode if self.xmode else "None")
-            s += "Y units: {:s}\n".format(self.ymode if self.ymode else "None")
+            s += "X units: {:s}\n".format(self.xmode or "None")
+            s += "Y units: {:s}\n".format(self.ymode or "None")
```
Comment (jvavrek) [inline becquerel/core/fitting.py:1439]: I dislike this change---it relies on a subtlety unique to python that not even many intermediate python programmers know, just for the sake of a little brevity. Could we disable this?

File: becquerel/parsers/iec1455.py [hunk 2]
```diff
@@ -95,7 +95,7 @@ def read(filename, verbose=False, cal_kwargs=None):
     energy_channel_pairs = []
     energy_res_pairs = []  # currently unused
     energy_eff_pairs = []  # currently unused
-    with Path(filename).open() as f:
+    with Path(filename).open(encoding="utf-8") as f:
```
Comment (jvavrek) [inline becquerel/parsers/iec1455.py:98]: is this really necessary?

File: tests/fitting_test.py [hunk 3]
```diff
@@ -31,7 +31,7 @@ def sim_data(x_min, x_max, y_func, num_x=200, binning="linear", **params):
         edges = np.linspace(x_min, x_max, num_x, dtype=float)
     elif binning == "sqrt":
         edges = np.linspace(np.sqrt(x_min), np.sqrt(x_max), num_x)
-        edges = edges**2
+        edges **= 2
```
Comment (jvavrek) [inline tests/fitting_test.py:34]: this exists? TIL

Comment (jvavrek) [reply tests/fitting_test.py:34]: not sure I like it

File: becquerel/core/fitting.py (hunk 1 above)
Comment (markbandstra) [reply becquerel/core/fitting.py:1439]: Your alternate idea: convert both to `f"{str(self.xymode)}"` for maximum conciseness.

Comment (markbandstra) [reply becquerel/core/fitting.py:1439]: This is almost certainly [FURB110](https://docs.astral.sh/ruff/rules/if-exp-instead-of-or-operator/)

**Comment by markbandstra** [comment]
I will redo this PR without implementing `ruff`'s [`preview` mode](https://docs.astral.sh/ruff/preview/) since it includes unstable rules.

File: becquerel/parsers/iec1455.py (hunk 2 above)
Comment (markbandstra) [reply becquerel/parsers/iec1455.py:98]: This is [`PLW1514`](https://docs.astral.sh/ruff/rules/unspecified-encoding/).

**Pull request closed without merging**